### PR TITLE
Complete overhaul of documentation

### PR DESCRIPTION
**Pull request by droidmonkey**
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" )
Complete overhaul of our documentation. All new format (asciidoctor) and images. Included SSH Agent documentation, KeeShare, and Browser Integration. Removed all old documentation sets.

Included man page documentation conversion to Asciidoctor, thanks to @sorairolake.

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )
[Getting Started Guide](https://staging.keepassxc.org/docs/KeePassXC_GettingStarted.html)
[User Guide](https://staging.keepassxc.org/docs/KeePassXC_UserGuide.html)

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Documentation (non-code change)

### CODE REVIEW

File: docs/topics/BrowserPlugin.adoc [hunk 1]
```diff
@@ -0,0 +1,86 @@
+= KeePassXC - Browser Plugin
+include::.sharedheader[]
+:imagesdir: ../images
+
+// tag::content[]
+== Setup Browser Integration
+The KeePassXC-Browser extension is installed within your web browser so that you can automatically pull usernames and passwords from KeePassXC and populate them directly into website fields. It is a very useful and secure extension that enhances your productivity while using KeePassXC. With this extension, You do not need to manually copy the data from your KeePassXC database and paste it into the website fields.
```
Comment (varjolintu) [inline docs/topics/BrowserPlugin.adoc:7]: Just a minor thing but: ", You do not need to" with lowercase "you"?

**Comment by varjolintu** [comment]
@droidmonkey Any plans to add the following:
- Database Settings -> Browser Integration
- Advanced Entry Options -> Browser Integration

**Comment by droidmonkey** [comment]
Added in the last push

File: docs/topics/SSHAgent.adoc [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    | o..  Eo. .      |
+    |  +o .. So       |
+    |o*o.o+ ..o       |
+    |Bo=+o.+.o .      |
+    |+oo+.++o         |
+    |. ..++ooo        |
+    +----[SHA256]-----+
+
+
+Now we can see two files wre generated:
```
Comment (hifi) [inline docs/topics/SSHAgent.adoc:93]: Typo.

File: docs/topics/SSHAgent.adoc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    $ ls -l keepassxc*
+    -rw------- 1 user group 2.6K Apr  5 07:36 keepassxc
+    -rw-r--r-- 1 user group  569 Apr  5 07:36 keepassxc.pub
+
+With KeePassXC you only need the first file listed.
+
+==== Generating a key on Windows with PuTTYgen
+Please read the manual on how to use PuTTYgen for details on generate a key: https://www.ssh.com/ssh/putty/putty-manuals/0.68/Chapter8.html#pubkey-puttygen. Once generated, you must save the key in OpenSSH format, follow the image below.
```
Comment (hifi) [inline docs/topics/SSHAgent.adoc:102]: SSH.com is not the official source for PuTTY.

File: docs/topics/SSHAgent.adoc [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+.Generating Key with PuttyGen
```
Comment (hifi) [inline docs/topics/SSHAgent.adoc:104]: Capitalization of PuTTYgen is wrong here and later. It should be `PuTTYgen` everywhere.

**Comment by droidmonkey** [comment]
Should be good to go now, going to merge this after CI finishes.

**Pull request merged**